test/unit-tests/broker-launch.test.js

```diff
@@ -8,23 +8,28 @@ const launchBroker = require('../../dashboard/index');
 // mock event emitter to simulate ChildProcess object
 const mockChildProcess = new EventEmitter();
 
-// sinon fakes for behaviour verification
-let fake = sinon.fake();
-let spawnArgs;
+// sinon spies for behaviour verification
+let unrefSpy = sinon.spy();
+let disconnectSpy = sinon.spy();
+
+// storing stubbed methods arguments for verification
+let spawnArgs, processArgs;
 
 describe('Broker daemonized launching', () => {
     before(() => {
         // stub cp.spawn, to avoid actual daemon launch
         sinon.stub(cp, 'spawn').callsFake((...args) => {
             spawnArgs = args;
 
-            mockChildProcess['unref'] = fake;
-            mockChildProcess['disconnect'] = fake;
+            mockChildProcess['unref'] = unrefSpy;
+            mockChildProcess['disconnect'] = disconnectSpy;
             return mockChildProcess;
```

Comment:
If we're not using any other property of event emitter other than `unref` and `disconnect`, we can just return these methods:

```suggestion
            return {
            	unref: unrefSpy,
            	disconnect: disconnectSpy
            };
```

Reply:
Actually we are using the ```.on``` method to listen events/messages via IPC.

Reply:
In that case, why not stub that as well? 🤔

Reply:
I did look into it, but if we extend the daemon IPC we might have to also add listeners for ```'stdin'``` ```'stderr'``` etc. and assert the daemon behavior on that basis. To uniquely assert the behavior of every such listener, we can individually ```emit``` that event in that particular ```it``` block and then assert the behavior of the response.